lib/rig_inbound_gateway_web/v1/metadata_controller.ex

```diff
@@ -0,0 +1,178 @@
+
+  There may be indexed metadata pairs defined. If a metadata set does not contain these indexed keys, it cannot be accepted.
+
+  ## Example
+
+  Metadata set that contains a user ID, locale and timezone:
+
+      {
+        "metadata": {
+          "userid": "9ab1bff2-a8d8-455c-b48a-50145d7d8e30",
+          "locale": "de-AT",
+          "timezone": "GMT+1"
+        }
+      }
+
+  In this example, the field "userid" might be indexed. In this case, the connection token would get associated with the value of the "userid" field (it will be possible to automatically take the user id from the JWT token (so that it cannot be changed by any other way) and index that; this will be serverside configuration).
+  In addition to that, the value of the "userid", "locale" and "timezone" fields would get associated to the connection token.
+
+  To sum it up: the connection token is indexed by default, while all other fields can be indexed by configuration.
+
+  So the association would look like so:
+
+      userid -> connection token
+
+      conection token ->
+        userid
+        locale
+        timezone
+
+  All indexed fields are then being propagated to all other instances, so they know where to find the corresponding metadata.
+
+  ## Example
+
+  A user connects to RIG 1 and sends the metadata fields from above to RIG 1. RIG 1 would propagate the connection token and the userid along with the RIG instance ID to all other RIG instances.
+  If someone were to access the metadata of a user from RIG 3, RIG 3 would know where to find the metadata, ask RIG 1 for it and return the metadata set.
+
+  ## Dirty testing with JWT
+
+      AUTH='Authorization:Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI5YWIxYmZmMi1hOGQ4LTQ1NWMtYjQ4YS01MDE0NWQ3ZDhlMzAiLCJuYW1lIjoiSm9obiBEb2UiLCJpYXQiOjE1Njg3MTMzNjEsImV4cCI6NDEwMzI1ODE0M30.kjiR7kFyOEeMJaY1zPCctut39eEWmKswUCNZdK5Q3-w'
+      META='{ "metadata": { "locale": "de-AT", "timezone": "GMT+2" } }'
+      http put ":4000/_rig/v1/connection/sse/${CONN_TOKEN}/metadata" <<< "$META" "$AUTH"
+
+  ## Dirty testing without JWT
+      META='{ "metadata": { "userid": "9ab1bff2-a8d8-455c-b48a-50145d7d8e30", "locale": "de-AT", "timezone": "GMT+2" } }'
+      http put ":4000/_rig/v1/connection/sse/${CONN_TOKEN}/metadata" <<< "$META"
+
+  """
+  @spec set_metadata(conn :: Plug.Conn.t(), params :: map) :: Plug.Conn.t()
+  def set_metadata(
+        %{method: "PUT"} = conn,
+        %{
+          "connection_id" => connection_id
+        }
+      ) do
+
+    conn
+    |> accept_only_req_for(["application/json"])
+    |> decode_metadata()
+    |> do_set_metadata(connection_id)
+  end
+
+  defp decode_metadata(conn) do
+    with {"application", "json"} <- content_type(conn),
+         {:ok, body, conn} <- BodyReader.read_full_body(conn),
+         {:ok, json} <- Jason.decode(body),
+         {:parse, %{"metadata" => metadata}} <- {:parse, json} do
+
+      conf = config()
+      jwt_fields = conf.jwt_fields
```

Comment:
`jwt_fields` is not a good name for this. It reads like "the fields of a JWT", while in fact this is far from that.

Reply:
why/how is this resolved?